### ISSUE TITLE
Selene should search for configuration files that begin with a dot

### ISSUE THREAD

**Issue by tomodachi94** [Kampfkarren/selene#469]
Selene should allow prefixing all of its configuration with a dot. [StyLua allows this.](https://github.com/JohnnyMorganz/StyLua#finding-the-configuration)

This would aid with having developer tooling configurations being distinguished from ordinary source files.

**Comment by tomodachi94** [comment]
For example, I would have a file, `.selene.toml`, that contains the same things as `selene.toml`. In addition, I would have a `.my-custom-stdlib.yaml` file that contains my custom standard library.

**Comment by tomodachi94** [comment]
Workaround: Specify the `--config` option when running `selene`.

**Comment by Kampfkarren** [comment]
`.selene.toml` seems fine, but I think standard libraries should just work with putting the dot in the std name itself, we already have a bunch of weird combinatorics with them

**Comment by Zeioth** [comment]
This is a convention of most formaters/linters. It would be very desireable.

**Comment by chriscerie** [comment]
Is there a stronger motivation for this? Introducing multiple ways to do the same thing is not great, and eslint is [moving away from such behavior](https://eslint.org/blog/2023/10/flat-config-rollout-plans/#eslintrc-removed-in-eslint-v10.0.0).

**Comment by LastTalon** [comment]
> eslint is [moving away from such behavior](https://eslint.org/blog/2023/10/flat-config-rollout-plans/#eslintrc-removed-in-eslint-v10.0.0).

ESLint is moving away from their old config format entirely. It's not really about prefixing with a `.`, which is usually just to treat the config as a hidden file.

**Comment by chriscerie** [comment]
> > eslint is [moving away from such behavior](https://eslint.org/blog/2023/10/flat-config-rollout-plans/#eslintrc-removed-in-eslint-v10.0.0).
> 
> ESLint is moving away from their old config format entirely. It's not really about prefixing with a `.`, which is usually just to treat the config as a hidden file.

I more meant there doesn’t seem to be a strong enough use case for eslint to support dot-prefixed configs in their new system. What’s different about lua where this would be worth a more complicated config resolution?

**Comment by chriscerie** [comment]
I'm going to close this for now as I don't see the benefits worth the added complexity. I'm not entirely opposed to reopening this in the future if there's a stronger motivation.

**Comment by Zeioth** [comment]
In case you are not familiar, on UNIX system (Linux, MacOS, and others), adding a dot in front of the file, make the file a hidden for file browsers. That way we avoid polluting our user projects.

**Comment by LastTalon** [comment]
> What’s different about lua where this would be worth a more complicated config resolution?

Right, as stated it has nothing to do with Lua and entirely to do with supporting filesystems that expect prefixing with a dot to hide files. This can easily be asked of any tooling that uses the file system. I can't find any documentation one way or the other with regard to eslint because, again, this is a new file name with an entirely new format and that coincidentally is removing the dot prefixing the old name. If they don't have plans for this with their new format I would fully expect issues to show up very quickly for it for the same reason, just as you find issues of the same nature for virtually every tool.

The better question is what makes selene unique to other tools in not wanting/needing to support hidden files on unix filesystems?

**Comment by chriscerie** [comment]
Thanks. When we see a mainstream tool move away from a convention it's natural to question why and it's better to document such considerations now than later.
